en/SUMMARY.md

```diff
@@ -610,10 +610,13 @@
     * [Consoles/Shells](debug/consoles.md)
       * [MAVLink Shell](debug/mavlink_shell.md)
       * [System Console](debug/system_console.md)
-    * [Embedded Debugging](debug/gdb_debugging.md)
-    * [SWD/JTAG Debug Interface](debug/swd_debug.md)
-    * [GDB SWD Debugging](debug/swd_gdb.md)
-    * [Eclipse JLink Debugging](debug/eclipse_jlink.md)
+    * [Debugging with GDB](debug/gdb_debugging.md)
```

Comment:
I like this structure, in particular that related topics are dropped one level so that the debug topic isn't quite so confronting. My assumptions are:
- We only use GDB debugging. Even if we use Eclipse say, it is GDB under the hood.
- All these probe topics are GDB specific in a PX4 context. What I'm getting at here is that Debugging with Eclipse will use these probes, so we need to make sure if you're working with eclipse you can still find them.

Reply:
Yes, I group it by UI.

The IDEs tend to hide all the probes and debuggers in their UI, so the casual developer may not even see a GDB console. The configuration is often also highly specific to the IDE, so it's probably best to keep them separate for now.

It's in my backlog to investigate how to integrate these GDB tools into VSCode, which also has several high-quality plugins for Cortex-M debugging. I would then add another VSCode specific guide and perhaps group it as "Debugging with IDEs".